src/lib.rs

```diff
@@ -345,6 +366,17 @@ impl<R: Read> Decoder<R> {
             // TODO(Kagami): interlacing, comment.
             match name {
                 b'W' => width = parse_bytes(value)?,
+                b'X' => comments.push(str::from_utf8(value).unwrap().to_owned()),
```

Comment:
Oh, I didn't know how to make it from_ascii instead of from_utf8. The only solution seems to be another create, which I don't feel is the right option. Lossy is the right option in my opinion if you want to remain with utf_8, what do you all think?

As for the parse error, I will fix the code to reflect that.

Reply:
According to ffmpeg
https://github.com/FFmpeg/FFmpeg/blob/n4.4/libavformat/yuv4mpegdec.c#L180
https://github.com/FFmpeg/FFmpeg/blob/n4.4/libavformat/yuv4mpegenc.c#L100
it's mostly non-standard extensions like colorspace/pixel format/color range, so lossy from_utf8 should work most of the time...

Reply:
How about
`b'X' => comments.push(String::from_utf8_lossy(value).into_owned())`
as a replacement for that line?

Reply:
Yeah, I think that's ok. But why comments is array? There should be only one per file header normally?

Reply:
There are multiple comments for each file. I think even FFMPEG produces y4m files with multiple comments. My application definitely generates multiple comments.